src/AutoMapper/Execution/TypeMapPlanBuilder.cs

```diff
@@ -522,6 +522,10 @@ private Expression BuildValueResolverFunc(PropertyMap propertyMap, Expression de
                     );
             }
 
+            valueResolverFunc = _typeMap.Profile.ValueTransformers
+                .Where(vt => vt.IsMatch(propertyMap))
+                .Aggregate(valueResolverFunc, (current, vtConfig) => ToType(ReplaceParameters(vtConfig.TransformerExpression, ToType(current, vtConfig.ValueType)), propertyMap.DestinationPropertyType));
```

Comment:
Can this change to having ValueTransformerConfiguration have function likeMapExpression?  Kinda like what IObjectMapper does now where it's an interface like IValueTransformerConfiguration.